internal/sql/pg/conn.go

```diff
@@ -145,6 +153,27 @@ func NewPool(ctx context.Context, cfg *PoolConfig) (*Pool, error) {
 		return defaultOnPgError(c, n) // automatically close any fatal errors (default we are overridding)
 	}
 
+	subscribers := syncmap.New[int64, chan<- string]()
+
+	pCfg.ConnConfig.OnNotice = func(pc *pgconn.PgConn, n *pgconn.Notice) {
+		// parse off the leading txid indicator. If successful, send the notice to the subscribers
+		txid, log, ok := containsTxid(n.Message)
+		if !ok {
+			return
+		}
+
+		sub, ok := subscribers.Get(txid)
+		if !ok {
+			return
+		}
+
+		select {
+		case sub <- log:
+		default:
+			logger.Warnf("txid %d subscriber is full, dropping notice", txid)
```

Comment:
Hopefully the presence or absence or even the content of the notice does not affect consensus in anyway. Can we engineer it to not affect consensus?

Reply:
Ah good point. I was debating this. Maybe we should just block here then?

Reply:
Possibly, if missing a log is a problem.  I guess it would be on account of cost?  What if the log contents didn't exactly  match on two different nodes?  Ideally we'd be robust to that too, unless this is typically something that affects consensus.

Reply:
The log contents should exactly match, as it will be part of consensus. Enforcing skipping logs on account of cost should happen at a higher level IMO (ideally with a gas pricing mechanism when the `notify` function is called, but for now it will simply stop reading logs if too many have been emitted).

Reply:
> The log contents should exactly match, as it will be part of consensus

Via cometbft's block hash, I assume?  (I don't know how this kind of emitted data is considered by the consensus engine.)  I don't expect _we'll_ be using these logs to affect our computed component of the app hash, just putting it in the TxResult to be retrieved by a tx query?

Reply:
> I don't expect we'll be using these logs to affect our computed component of the app hash, just putting it in the TxResult to be retrieved by a tx query? i.e. it's not going to change application state, only blockchain data.

Exactly

Reply:
Circling back on this, @charithabandi discovered that cometbft does not consider logs or events in the deterministic part of tx results, so block hash should be unaffected regardless.